python/example/network_ring_gpu.py

```diff
@@ -123,7 +123,7 @@ def global_properties(self, kind):
 # (11) Set up the hardware context
 # gpu_id set to None will not use a GPU.
 # gpu_id=0 instructs Arbor to the first GPU present in your system
-context = arbor.context(threads="avail_threads", gpu_id=None)
+context = arbor.context(1, gpu_id=None)
```

Comment:
Why exactly do we opt to use one thread here?

Reply:
Since an integer number must be provided when the nondefault constructor is used (`"avail_threads"` has been removed), and 1 is as good as any.

Reply:
Oh dear, I missed the removal. Please keep it in. Reasoning: We shouldn't be doing something the user cannot ask for.

Reply:
Before I do that: it requires a combinatorial explosion of constructors, especially if we want to be able to set it in any combination with the other options (maybe we should switch to kwargs only?) due to the different type. I considered `py::any()`, but that didn't make it any more brief.

Reply:
Those who dabble in dynamically typed languages are damned to type `type_of(...)` into all eternity.
kw_args is fine in Python, I think (I can never remember the ordering anyhow). C++ should be ok with
taking an integral type for the thread count (which can be defaulted to `arbenv::thread_concurrency`)
and the GPU id.